tests/core/test_asan_memchr.c

```diff
@@ -1,5 +1,5 @@
 #include <string.h>
 
 int main() {
-  memchr("hello", 'z', 7);
+  return (int)memchr("hello", 'z', 7);
```

Comment:
why does LTO break without this?

Reply:
I think at higher optimization levels the entire call to memchr is eliminated which make asan happy.   I think this is either because memchr is a builtin and llvm can reason about its semantics (assuming `-fno-builtins` is not passed).   It could also be that the compiler can see that the read is out of bounds which is undefined behaviour and then do whatever strange optimizations it wants to.